packages/design-picker/src/components/mshots-image/index.tsx

```diff
@@ -1,32 +1,42 @@
 /**
  * External dependencies
  */
-import React, { useState, useLayoutEffect } from 'react';
+import React, { useState, useEffect, useLayoutEffect } from 'react';
 import classnames from 'classnames';
 import { addQueryArgs } from '@wordpress/url';
 
 /**
  * Style dependencies
  */
+import { isEnabled } from '@automattic/calypso-config';
 import './style.scss';
 
 interface MShotsImageProps {
 	url: string;
 	alt: string;
 	'aria-labelledby': string;
 	options: MShotsOptions;
+	scrollable?: boolean;
 }
 
 export type MShotsOptions = {
 	vpw: number;
 	vph: number;
 	w: number;
-	h: number;
+	h?: number;
+	screen_height?: number;
 };
 
+const cacheBuster = isEnabled( 'gutenboarding/bust-mshots-cache' )
+	? `&cache_buster=${ Date.now() }`
+	: '';
+
 export function mshotsUrl( url: string, options: MShotsOptions, count = 0 ): string {
-	const mshotsUrl = 'https://s0.wp.com/mshots/v1/';
-	const mshotsRequest = addQueryArgs( mshotsUrl + encodeURIComponent( url ), {
+	const mshotsUrl = isEnabled( 'gutenboarding/local-mshots' )
+		? 'http://127.0.0.1:8000/mshots/v1/'
+		: 'https://s0.wp.com/mshots/v1/';
+
+	const mshotsRequest = addQueryArgs( mshotsUrl + encodeURIComponent( url + cacheBuster ), {
```

Comment:
`url + cacheBuster` works fine now, but perhaps we should be more defensive in case things change one day up the call stack such that `url` no longer includes other query params.

It's more verbose, but something like this feels more resilient to future changes:
```
const urlWithCacheBuster = !cacheBuster ? url : addQueryArgs( url, { cache_buster: cacheBuster } );
const mshotsRequest = addQueryArgs( mshotsUrl + encodeURIComponent( urlWithCacheBuster ), {
   // ...
} );
```

Reply:
Oh, nice one! That could have been really annoying if it ever went wrong: https://github.com/Automattic/wp-calypso/pull/51261/commits/2422655d88d0456294285559a6cb1e831fd427f6